code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -115,6 +115,36 @@
 	handle_reactions()
 	return total_transfered
 
+/datum/reagents/proc/remove_any_but(var/amount=1, var/reagent_to_ignore) //removes all other than the designated ignored reagent
```

Comment:
Default variables should not be before non-default variables.

Reply:
done, moved reagent to ignore first